apps/dashboard/app/(app)/settings/billing/stripe/success/page.tsx

```diff
@@ -72,19 +76,64 @@ export default async function StripeSuccess(props: Props) {
     );
   }
 
-  const isChangingPlan = new_plan && new_plan !== ws.plan;
+  const isUpgradingPlan = new_plan && new_plan !== ws.plan && new_plan === "pro";
+  const h = headers();
 
-  await db
-    .update(schema.workspaces)
-    .set({
-      stripeCustomerId: customer.id,
-      stripeSubscriptionId: session.subscription as string,
-      trialEnds: null,
-      ...(isChangingPlan ? { plan: new_plan } : {}),
-    })
-    .where(eq(schema.workspaces.id, ws.id));
+  await db.transaction(async (tx) => {
+    await tx
+      .update(schema.workspaces)
+      .set({
+        stripeCustomerId: customer.id,
+        stripeSubscriptionId: session.subscription as string,
+        trialEnds: null,
+        ...(isUpgradingPlan
+          ? {
+              plan: new_plan,
+              planChanged: new Date(),
+              subscriptions: defaultProSubscriptions(),
+              planDowngradeRequest: null,
+            }
+          : {}),
+      })
+      .where(eq(schema.workspaces.id, ws.id));
 
-  if (isChangingPlan) {
+    if (isUpgradingPlan) {
+      await insertAuditLogs(tx, {
+        workspaceId: ws.id,
+        actor: { type: "user", id: user.id },
+        event: "workspace.update",
+        description: "Changed plan to 'pro'",
+        resources: [
+          {
+            type: "workspace",
+            id: ws.id,
+          },
+        ],
+        context: {
+          location: h.get("user-agent") ?? "unknown",
+          userAgent: h.get("x-forwarded-for") ?? process.env.VERCEL_REGION ?? undefined,
+        },
+      });
+      await ingestAuditLogsTinybird({
+        workspaceId: ws.id,
+        actor: { type: "user", id: user.id },
+        event: "workspace.update",
+        description: "Changed plan to 'pro'",
+        resources: [
+          {
+            type: "workspace",
+            id: ws.id,
+          },
+        ],
+        context: {
+          location: h.get("user-agent") ?? "unknown",
+          userAgent: h.get("x-forwarded-for") ?? process.env.VERCEL_REGION ?? undefined,
+        },
+      });
+    }
+  });
+
+  if (isUpgradingPlan) {
```

Comment:
nit: we could just merge the two `if` clauses, right?

Reply:
I kept the PostHog call outside of the transaction call intentionally. If we want to merge the two if clauses, I can move the audit logging outside of the transaction too, but it looks like we wanted audit logging with the transaction. 

the two clauses are there for one within the transaction and one outside the transaction